sgx-tools/dcap_linux.go

```diff
@@ -0,0 +1,358 @@
+package main // import "github.com/inclavare-containers/sgx-tools"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"github.com/go-restruct/restruct"
+	"github.com/golang/protobuf/proto"
+	"github.com/inclavare-containers/rune/libenclave/intelsgx"
+	pb "github.com/inclavare-containers/sgx-tools/proto"
+	"github.com/sirupsen/logrus"
+	"net"
+)
+
+const (
+	aesmdSocket         = "/var/run/aesmd/aesm.socket"
+	attKeyIdLength      = 256
+	nonceLength         = 16
+	qeReportInfoLength  = 960
+	timeOut             = 15000000
+	defaultDcapAttKeyId = "0000000020008c4f5775d796503e96137f77c68a829a0056ac8ded70140b081b094490c57bff0000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+)
+
+func dialAesmd() (*net.UnixConn, error) {
+	addr, err := net.ResolveUnixAddr("unix", aesmdSocket)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := net.DialUnix("unix", nil, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+func transmitAesmd(conn *net.UnixConn, req *pb.AesmServiceRequest) ([]byte, error) {
+	rdata, err := proto.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	msgSize := uint32(len(rdata))
+	byteBuf := bytes.NewBuffer([]byte{})
+	binary.Write(byteBuf, binary.LittleEndian, &msgSize)
+	if _, err = conn.Write(byteBuf.Bytes()); err != nil {
+		return nil, err
+	}
+
+	if _, err = conn.Write(rdata); err != nil {
+		return nil, err
+	}
+
+	rdata = append(rdata[:4])
```

Comment:
We should avoid to use any hardcode value. I know this is the code appearing in intelsgx package. But this 4-byte indicates the fixed length field used to describe the following pb message length. So it is necessary to call it rawMessageLength or similar.

Reply:
> We should avoid to use any hardcode value. I know this is the code appearing in intelsgx package. But this 4-byte indicates the fixed length field used to describe the following pb message length. So it is necessary to call it rawMessageLength or similar.

OK.